include/rocksdb/env.h

```diff
@@ -329,6 +330,21 @@ class Env {
   // Returns the ID of the current thread.
   virtual uint64_t GetThreadID() const;
 
+  // Load Env private metadata
+  virtual Status LoadPrivateMetadata(std::string fname, void* metadata) {
+    return Status::OK();
+  }
+
+  // Encode private metadata used by the storage backed implementing Env
+  static void EncodePrivateMetadata(std::string* priv, void* metadata);
+
+  // Retrieve and decode private metadata from the storage backend implementing
+  // Env
+  static void* DecodePrivateMetadata(Slice* input);
+
+  // Free private metadata
+  static void FreePrivateMetadata(void* metadata);
```

Comment:
@javigon do you want to address the comment of not using static Env functions? I suggest we store a structure in FileMetaData containing both of Env and the handle pointer. Is it OK?